apps/staking/src/api.ts

```diff
@@ -0,0 +1,205 @@
+  MOCK_DATA.walletAmount += amount;
+};
+
+export const claim = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  MOCK_DATA.total += MOCK_DATA.availableRewards;
+  MOCK_DATA.availableRewards = 0n;
+};
+
+export const stakeGovernance = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  MOCK_DATA.governance.warmup += amount;
+};
+
+export const cancelWarmupGovernance = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  MOCK_DATA.governance.warmup -= amount;
+};
+
+export const unstakeGovernance = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  MOCK_DATA.governance.staked -= amount;
+  MOCK_DATA.governance.cooldown += amount;
+};
+
+export const delegateIntegrityStaking = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  publisherKey: string,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  const publisher = MOCK_DATA.integrityStakingPublishers.find(
+    (publisher) => publisher.publicKey === publisherKey,
+  );
+  if (publisher) {
+    publisher.positions ||= {};
+    publisher.positions.warmup = (publisher.positions.warmup ?? 0n) + amount;
+  } else {
+    throw new Error(`Invalid publisher key: "${publisherKey}"`);
+  }
+};
+
+export const cancelWarmupIntegrityStaking = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  publisherKey: string,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  const publisher = MOCK_DATA.integrityStakingPublishers.find(
+    (publisher) => publisher.publicKey === publisherKey,
+  );
+  if (publisher) {
+    if (publisher.positions?.warmup) {
+      publisher.positions.warmup -= amount;
+    }
+  } else {
+    throw new Error(`Invalid publisher key: "${publisherKey}"`);
+  }
+};
+
+export const unstakeIntegrityStaking = async (
+  _connection: Connection,
+  _wallet: WalletContextState,
+  publisherKey: string,
+  amount: bigint,
+): Promise<void> => {
+  await new Promise((resolve) => setTimeout(resolve, MOCK_DELAY));
+  const publisher = MOCK_DATA.integrityStakingPublishers.find(
+    (publisher) => publisher.publicKey === publisherKey,
+  );
+  if (publisher) {
+    if (publisher.positions?.staked) {
+      publisher.positions.staked -= amount;
+      publisher.positions.cooldown =
+        (publisher.positions.cooldown ?? 0n) + amount;
+    }
+  } else {
+    throw new Error(`Invalid publisher key: "${publisherKey}"`);
+  }
+};
```

Comment:
@keyvankhademi @guibescos here are the mock APIs which we will need to replace with correct on-chain calls.

I think the general signatures are reasonable except perhaps `loadData`.  I opted to make that a single API that returns everything, rather than splitting up governance/publishers/publisher stakes or similar, because by making it a single API it makes it much more trivial to just refresh the whole data set every time any transfer is completed and not have to worry about how things affect each other (for instance if I withdraw a bunch of tokens, how does that affect the counts that show up in the governance & integrity sections).  It's not that hard to manage those things, but it's marginally messier and adds more logic on the frontend.  So, if there's a good reason to split up that API then I can update accordingly, but if not I recommend just keeping it all in one API.